modules/adagioAnalyticsAdapter.js

```diff
@@ -24,24 +42,287 @@ function canAccessTopWindow() {
+};
+
+/**
+ * Send a new beacon to Adagio. It increment the version of the beacon.
+ * @param {string} auctionId
+ * @param {string} adUnitCode
+ */
+function newBeacon(auctionId, adUnitCode) {
+  cache.updateAuction(auctionId, adUnitCode, {
+    v: (cache.getAuction(auctionId, adUnitCode).v || 0) + 1
+  });
+  sendRequest(cache.getAuction(auctionId, adUnitCode));
+};
+
+/**
+ * END UTILS FUNCTIONS
+*/
+
+/**
+ * HANDLERS
+ * - handlerAuctionInit
+ * - handlerBidResponse
+ * - handlerBidWon
+ * - handlerAdRender
+ *
+ * Each handler is called when the event is fired.
+*/
+
+function handlerAuctionInit(event) {
+  const w = getCurrentWindow();
+
+  const prebidAuctionId = event.auctionId;
+  const adUnitCodes = removeDuplicates(event.adUnitCodes, adUnitCode => adUnitCode);
+
+  // Check if Adagio is on the bid requests.
+  // If not, we don't need to track the auction.
+  const adagioBidRequest = event.bidderRequests.find(bidRequest => isAdagio(bidRequest.bidderCode));
+  if (!adagioBidRequest) {
+    logInfo(`Adagio is not on the bid requests for auction '${prebidAuctionId}'`)
+    return;
+  }
+
+  cache.auctions[prebidAuctionId] = {};
+
+  adUnitCodes.forEach(adUnitCode => {
+    const adUnits = event.adUnits.filter(adUnit => adUnit.code === adUnitCode);
+
+    // Get all bidders configures for the ad unit.
+    const bidders = removeDuplicates(
+      adUnits.map(adUnit => adUnit.bids.map(bid => ({bidder: bid.bidder, params: bid.params}))).flat(),
+      bidder => bidder.bidder
+    );
+
+    // Check if Adagio is configured for the ad unit.
+    // If not, we don't need to track the ad unit.
+    const adagioBidder = bidders.find(bidder => isAdagio(bidder.bidder));
+    if (!adagioBidder) {
+      logInfo(`Adagio is not configured for ad unit '${adUnitCode}'`);
+      return;
+    }
+
+    // Get all media types and banner sizes configured for the ad unit.
+    const mediaTypes = adUnits.map(adUnit => adUnit.mediaTypes);
+    const mediaTypesKeys = removeDuplicates(
+      mediaTypes.map(mediaTypeObj => Object.keys(mediaTypeObj)).flat(),
+      mediaTypeKey => mediaTypeKey
+    ).map(mediaType => getMediaTypeAlias(mediaType)).sort();
+    const bannerSizes = removeDuplicates(
+      mediaTypes.filter(mediaType => mediaType.hasOwnProperty(BANNER))
+        .map(mediaType => mediaType[BANNER].sizes.map(size => size.join('x')))
+        .flat(),
+      bannerSize => bannerSize
+    ).sort();
+
+    // Get all Adagio bids for the ad unit from the bidRequest.
+    // If no bids, we don't need to track the ad unit.
+    const adagioAdUnitBids = adagioBidRequest.bids.filter(bid => bid.adUnitCode === adUnitCode);
+    if (deepAccess(adagioAdUnitBids, 'length', 0) <= 0) {
+      logInfo(`Adagio is not on the bid requests for ad unit '${adUnitCode}' and auction '${prebidAuctionId}'`)
+      return;
+    }
+    // Get Adagio params from the first bid.
+    // We assume that all Adagio bids for a same adunit have the same params.
```

Comment:
That's not the case. Some params are the same (`organization_id`), some are not (`placement`)

Reply:
oops. I did not see the "same adunit". My bad